<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Lefser finds features that have greatest differences between classes. • lefser</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Lefser finds features that have greatest differences between classes.">
<meta property="og:description" content="lefser">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-90028752-3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-90028752-3');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">lefser</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/lefser.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/lefser/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="lefser_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Lefser finds features that have greatest differences between classes.</h1>
                        <h4 class="author"><div class="line-block">Asya Khleborodova, Ludwig Geistlinger, and Levi Waldron<br><small>School of Public Health, City University of New York</small>
</div></h4>
            
            <h4 class="date">2020-10-22</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/lefser/blob/master/vignettes/lefser.Rmd"><code>vignettes/lefser.Rmd</code></a></small>
      <div class="hidden name"><code>lefser.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Lefser is metagenomic biomarker discovery tool that is based on <a href="https://huttenhower.sph.harvard.edu/galaxy/">LEfSe</a> tool and is published by <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3218848/">Huttenhower et al. 2011</a>. <code>Lefser</code> is the R implementation of the <code>LEfSe</code> method.</p>
<p>Using statistical analyses, <code>lefser</code> compares microbial populations of healthy and diseased subjects to discover differencially expressed microorganisms. <code>Lefser</code> than computes effect size, which estimates magnitude of differential expression between the populations for each differentially expressed microorganism. Subclasses of classes can also be assigned and used within the analysis.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>To install Bioconductor and the <code>lefser</code> package, run the following commands.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"lefser"</span><span class="op">)</span></pre></div>
<p>Then load the <code>lefser</code> package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/lefser">lefser</a></span><span class="op">)</span></pre></div>
</div>
<div id="overview-and-example-use-of-lefser" class="section level1">
<h1 class="hasAnchor">
<a href="#overview-and-example-use-of-lefser" class="anchor"></a>Overview and example use of <code>lefser</code>
</h1>
<p>The <code>lefser</code> function can be used with a <code>SummarizedExperiment</code>.</p>
<p>Load the <code>zeller14</code> example dataset and exclude ‘adenoma’ conditions.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">zeller14</span><span class="op">)</span>
<span class="va">zeller14</span> <span class="op">&lt;-</span> <span class="va">zeller14</span><span class="op">[</span>, <span class="va">zeller14</span><span class="op">$</span><span class="va">study_condition</span> <span class="op">!=</span> <span class="st">"adenoma"</span><span class="op">]</span></pre></div>
<p>Note. <code>lefser</code> supports only two-group contrasts.</p>
<p>The <code>colData</code> in the <code>SummarizedExperiment</code> dataset contains the grouping column <code>study_condition</code> which includes the ‘control’ and ‘CRC’ groups.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">zeller14</span><span class="op">$</span><span class="va">study_condition</span><span class="op">)</span></pre></div>
<pre><code>## 
## control     CRC 
##      66      91</code></pre>
<p>There can be subclasses in each group condition. In the example dataset we include <code>age_category</code> as a subclass of <code>study_condition</code> which includes ‘adults’ and ‘seniors’. This variable will correspond to the <code>blockCol</code> input argument.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">zeller14</span><span class="op">$</span><span class="va">age_category</span><span class="op">)</span></pre></div>
<pre><code>## 
##  adult senior 
##     91     66</code></pre>
<p>We can create a contingency table for the two categorical variables.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">zeller14</span><span class="op">$</span><span class="va">age_category</span>, <span class="va">zeller14</span><span class="op">$</span><span class="va">study_condition</span><span class="op">)</span></pre></div>
<pre><code>##         
##          control CRC
##   adult       46  45
##   senior      20  46</code></pre>
<p>We can now use the <code>lefser</code> function. It provides results as a <code>data.frame</code> with the names of selected microorganisms and their effect size.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lefser.html">lefser</a></span><span class="op">(</span><span class="va">zeller14</span>, groupCol <span class="op">=</span> <span class="st">"study_condition"</span>, blockCol <span class="op">=</span> <span class="st">"age_category"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></pre></div>
<pre><code>##                                                                                                                                                           Names
## 1                                                                                                     `k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales`
## 2                                   `k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Ruminococcaceae|g__Ruminococcus|s__Ruminococcus_sp_5_1_39BFAA`
## 3                  `k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Ruminococcaceae|g__Ruminococcus|s__Ruminococcus_sp_5_1_39BFAA|t__GCF_000159975`
## 4                                             `k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Eubacteriaceae|g__Eubacterium|s__Eubacterium_hallii`
## 5                            `k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Eubacteriaceae|g__Eubacterium|s__Eubacterium_hallii|t__GCF_000173975`
## 6 `k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Lactobacillaceae|g__Lactobacillus|s__Lactobacillus_ruminis|t__Lactobacillus_ruminis_unclassified`
##      scores
## 1 -5.969502
## 2 -5.583681
## 3 -5.583681
## 4 -5.517931
## 5 -5.517931
## 6 -5.327371</code></pre>
</div>
<div id="visualizing-results-with-lefserplot" class="section level1">
<h1 class="hasAnchor">
<a href="#visualizing-results-with-lefserplot" class="anchor"></a>Visualizing results with <code>lefserPlot</code>
</h1>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="fu"><a href="../reference/lefserPlot.html">lefserPlot</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></pre></div>
<p><img src="lefser_files/figure-html/unnamed-chunk-8-1.png" width="700" style="display: block; margin: auto;"></p>
<div id="sessioninfo" class="section level2">
<h2 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>sessionInfo</h2>
<details><div class="sourceCode" id="cb13"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></pre></div>
<pre><code>## R version 4.0.3 Patched (2020-10-13 r79342)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.1 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] lefser_0.99.9               SummarizedExperiment_1.19.9
##  [3] Biobase_2.49.1              GenomicRanges_1.41.6       
##  [5] GenomeInfoDb_1.25.11        IRanges_2.23.10            
##  [7] S4Vectors_0.27.13           BiocGenerics_0.35.4        
##  [9] MatrixGenerics_1.1.3        matrixStats_0.57.0         
## [11] BiocStyle_2.17.1           
## 
## loaded via a namespace (and not attached):
##  [1] mvtnorm_1.1-1          lattice_0.20-41        zoo_1.8-8             
##  [4] assertthat_0.2.1       rprojroot_1.3-2        digest_0.6.25         
##  [7] R6_2.4.1               backports_1.1.10       evaluate_0.14         
## [10] ggplot2_3.3.2          pillar_1.4.6           zlibbioc_1.35.0       
## [13] rlang_0.4.7            multcomp_1.4-14        Matrix_1.2-18         
## [16] rmarkdown_2.4          pkgdown_1.6.1          labeling_0.3          
## [19] desc_1.2.0             splines_4.0.3          stringr_1.4.0         
## [22] RCurl_1.98-1.2         munsell_0.5.0          DelayedArray_0.15.12  
## [25] compiler_4.0.3         xfun_0.17              pkgconfig_2.0.3       
## [28] systemfonts_0.3.2      libcoin_1.0-6          htmltools_0.5.0       
## [31] tidyselect_1.1.0       tibble_3.0.3           GenomeInfoDbData_1.2.3
## [34] bookdown_0.20          coin_1.3-1             codetools_0.2-16      
## [37] crayon_1.3.4           dplyr_1.0.2            MASS_7.3-53           
## [40] bitops_1.0-6           grid_4.0.3             gtable_0.3.0          
## [43] lifecycle_0.2.0        magrittr_1.5           scales_1.1.1          
## [46] stringi_1.5.3          farver_2.0.3           XVector_0.29.3        
## [49] fs_1.5.0               ellipsis_0.3.1         ragg_0.3.1            
## [52] vctrs_0.3.4            generics_0.0.2         sandwich_2.5-1        
## [55] TH.data_1.0-10         tools_4.0.3            glue_1.4.2            
## [58] purrr_0.3.4            survival_3.2-7         yaml_2.2.1            
## [61] colorspace_1.4-1       BiocManager_1.30.10    memoise_1.1.0         
## [64] knitr_1.30             modeltools_0.2-23</code></pre>
</details>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Asya Khleborodova.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
